x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java

```diff
@@ -3194,14 +3203,612 @@ public void testValidateOwnerUserRoleDescriptorsWithWorkflowsRestriction() {
             randomList(1, 3, () -> randomAlphaOfLengthBetween(3, 5)),
             requestRoleDescriptors,
             Map.of(),
-            ApiKeyTests.randomFutureExpirationTime()
+            ApiKeyTests.randomFutureExpirationTime(),
+            null
         );
         final PlainActionFuture<BulkUpdateApiKeyResponse> updateFuture = new PlainActionFuture<>();
         service.updateApiKeys(authentication, updateRequest, userRoleDescriptorsWithWorkflowsRestriction, updateFuture);
         final IllegalArgumentException e2 = expectThrows(IllegalArgumentException.class, createFuture::actionGet);
         assertThat(e2.getMessage(), containsString("owner user role descriptors must not include restriction"));
     }
 
+    public void testApiKeyDocSerializationWithCertificateIdentity() throws IOException {
```

Comment:
The coverage of these tests is great. I wonder if the tests across `ApiKeyRestIT`, `ApiKeyServiceTests` and `ApiKeyIntegTests` could be consolidated and maybe more focused? We're testing the same thing a couple of times. Not sure exactly how that would be divided but happy to help out over zoom.